### ISSUE TITLE
Color method (toHSV) not working

### ISSUE THREAD

**Issue by Jazzelhawk** [INPStarfall/Starfall#311]
When trying to use the toHSV method on a color is says its a nil value. I tried adding a test function to the color library and that says nil value too. I think it was something to do with #284 

This is the error if it helps:

```
stack traceback:
    SF:generic:4: in function 'func'
    addons/starfall/lua/starfall/instance.lua:37: in function <addons/starfall/lua/starfall/instance.lua:36>
    [C]: in function 'xpcall'
    addons/starfall/lua/starfall/instance.lua:69: in function 'runWithOps'
    addons/starfall/lua/starfall/instance.lua:126: in function 'initialize'
    addons/starfall/lua/entities/starfall_screen/cl_init.lua:144: in function 'CodeSent'
    addons/starfall/lua/entities/starfall_screen/cl_init.lua:32: in function 'func'
    lua/includes/modules/net.lua:32: in function <lua/includes/modules/net.lua:17>
```

**Comment by awilliamson** [comment]
Can you include some test code so we can reproduce this issue. I'm wondering why all other colour methods work except toHSV, doesn't sound like a 'base' problem.

**Comment by Jazzelhawk** [comment]
`print( Color( 255, 255, 255 ):toHSV() )`

And there's only one colour method which isn't a metamethod.

I'm pretty sure the problem is in the __index metamethod. The __index method is only called when the key to a table doesnt exist. In [this line of code](https://github.com/INPStarfall/Starfall/blob/03f90b74b8724c3f1a373bbfa58233e01dc1cbc4/lua/starfall/libs_sh/color.lua#L46) you are basically just trying to get the key from the same table (which wont be there because __index was called).

**Comment by awilliamson** [comment]
If the index is ( 1, 2, 3, 4, r, g, b, a ) it will rawget ( which doesn't use __index metamethods for access ) on the unwrapped colour table. Any other indexes will use rawget ( again, no use of __index metamethod ) on the wrapped color table. This allows people to add custom values to the wrapped table, without changing the underlying structure.

I fail to see what the issue is here, I can't reproduce the problem unless you give full sample code. Ensure you are on the correct branch is all I can say ( Perhaps I could be on the wrong branch ). I'll double check.

**Comment by awilliamson** [comment]
Fixed by 2a19dac676e1cd2af9b30b2754f92d75aacf1a3c
